### ISSUE TITLE
Implement a formal RFC process

### ISSUE THREAD

**Issue by dangerbarber** [dzgames/TimberWolf#2]
To manage the endlessly evolving goals of this project, and try to nail them to the wall a bit better, and to prevent the patterns and APIs used from being entirely at the whims of whoever happens to be implementing a feature, I think that it would be very important (especially this early in the process of developing the engine), to write up a strict set of rules. These rules should outline the objectives of the engine, a set of standards for what patterns the code should follow, processes for changing APIs, processes for adding new features, etc. These rules also ought to be a living document of sorts, so they can evolve with time as needs become clearer, or the field evolves, or whatever else might happen going forward. I'd also like the RFC process to be governed by a well-written RFC, and I'm hoping that such an RFC process can be found in a pre-written and battle-tested form.

TL;DR: This request includes three clear phases at this time.
- Spike/research existing RFC processes
- Implement an RFC for the RFC process based on the above research
- Implement RFCs for the engine's objectives, which will guide decisions going forward
